packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -13,29 +13,43 @@
  *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
  */
 
-import { Dispatch, SetStateAction, useState } from 'react';
+import { Dispatch, SetStateAction, useState, useEffect } from 'react';
 import {
+  useDeleteAnimalBatchesMutation,
+  useDeleteAnimalsMutation,
   useRemoveAnimalBatchesMutation,
   useRemoveAnimalsMutation,
 } from '../../../store/api/apiSlice';
 import { toLocalISOString } from '../../../util/moment';
 import { parseInventoryId } from '../../../util/animal';
-import { FormFields } from '../../../components/Animals/RemoveAnimalsModal';
+import { CREATED_IN_ERROR_ID, FormFields } from '../../../components/Animals/RemoveAnimalsModal';
+import useMutations from '../../../hooks/api/useMutations';
+import { enqueueErrorSnackbar, enqueueSuccessSnackbar } from '../../Snackbar/snackbarSlice';
+import { TFunction } from 'i18next';
 import { AnimalOrBatchKeys } from '../types';
 
 const useAnimalOrBatchRemoval = (
   selectedInventoryIds: string[],
   setSelectedInventoryIds: Dispatch<SetStateAction<string[]>>,
+  dispatch: Dispatch<any>,
+  t: TFunction,
```

Comment:
Let me ask a question I've always had... (sorry Duncan)
@antsgar I'm not sure if I should pass `dispatch`/`t` or call `useDispatch()` /`useTranslation()` in this hook, or if I should pass `onAddAnimalsSuccess`, `onAddAnimalsFail`, `onAddBatchesSuccess` etc.
Do you have any advice on a good approach to take?

Reply:
If the effect after removing animals is always going to be the same regardless of where we reuse this logic (which won't be reused at the moment I think so it's hard to tell), I think it's easier to encapsulate those side effects within the hook itself and let the component not have to worry about it. If it's dependent on where in the UI the hook is called, I'd pass the handlers as props.

The only minor caveat is I'd call `useDispatch` and `useTranslation` in the hook itself as well so that each component doesn't have the extra work of calling that and passing that through.